README.md

```diff
@@ -13,312 +13,252 @@ to [dev.poktroll.com/category/actors](https://dev.poktroll.com/category/actors).
+It offers these methods:
 
-As a concrete example, the `Client` struct is exported directly from the `net/http` package
+- `GetApplication`: Retrieves application information for a specified application address.
+- `GateAllApplications`: Retrieves all available applications on the network.
+- `GetApplicationsDelegatingToGateway`: Retrieves applications delegating to the gateway.
 
-```go
-type Client struct {
-    // Transport specifies the mechanism by which individual
-    // HTTP requests are made.
-    // If nil, DefaultTransport is used.
-    Transport RoundTripper
-```
+The `ApplicationClient` depends on the `poktroll` application query client,
+which provides methods to fetch corresponding information from the Pocket network.
 
-Note that the above `Client` struct has multiple public methods, yet no code exists to force it to fulfill a specific interface.
+Refer to [application.go](https://github.com/pokt-network/shannon-sdk/blob/main/application.go)
+for detailed information.
 
-As a concrete example of keeping interfaces on the consumer side, the above `Client` consumes a `RoundTripper` interface.
-As the godoc page for `net/http` specifies: `For control over proxies, TLS configuration, keep-alives, compression, and other settings, create a Transport`:
+#### Application Ring
 
-```go
-tr := &http.Transport{
-    MaxIdleConns:       10,
-    IdleConnTimeout:    30 * time.Second,
-    DisableCompression: true,
-}
-client := &http.Client{Transport: tr}
-```
+The `ApplicationRing` retrieves delegated gateways from `Application`s and generates
+ring signatures from any actor in the ring.
 
-Example of a helper function used for overriding the default `RoundTripper`:
+It offers the following method:
 
-```func NewFileTransport(fs FileSystem) RoundTripper```
+- `GetRing`: Retrieves the ring associated with the application.
 
-#### Exposed Concrete Types
+The `ApplicationRing` relies on the `PublicKeyFetcher` interface, which requires
+implementations to fetch the public key of the associated application.
 
-Each file (in the top level directory) will have a client implemented and returned
-as a concrete struct.
+**Note**: The `AccountClient` implements the `PublicKeyFetcher` interface and can
+be used as a default implementation.
 
-For example, `ApplicationClient` is a `struct` that will be returned by `application.go`
-rather than an interface.
+Refer to [application.go](https://github.com/pokt-network/shannon-sdk/blob/main/application.go)
+for detailed information.
 
-#### sdk.go
+#### Block Client
 
-**NOTE: If you are reading this and the documentation is outdated, please update it!**
+The `BlockClient` fetches block information (e.g., block height) from the Pocket network.
 
-The `sdk.go` is a **TEMPORARY** that needs to be split file needs to be split into
-`application.go`, `supplier.go`, etc...
+It provides the following method:
```

Comment:
I think this style of documentation is prone to become outdated. Wdyt about adding a TODO to find a way to reference/integrate/deduplicate godoc comments?